.github/workflows/analyze.yml

```diff
@@ -21,11 +24,6 @@ jobs:
         # a pull request then we can checkout the head.
         fetch-depth: 2
 
-    # If this run was triggered by a pull request event, then checkout
-    # the head of the pull request instead of the merge commit.
-    - run: git checkout HEAD^2
-      if: ${{ github.event_name == 'pull_request' }}
-
```

Comment:
Why are these changes necessary? Incl. removing the on push branches/tags (L5-6), and adding permissions (L14-18).

Reply:
just merging @waltjones changes from PR #403